run_reg_test.py

```diff
@@ -0,0 +1,351 @@
+    if not (test_dir / "qor_geomean.txt").is_file():
+        print("QoR results do not exist ({}/qor_geomean.txt)".format(str(test_dir)))
+        return 1
+    print("=" * 121)
+    print("\t" * 6, end="")
+    print("{} QoR Results".format(reg_test))
+    print("=" * 121)
+    with (test_dir / "qor_geomean.txt").open("r") as results:
+        data = OrderedDict()
+        data["revision"] = [8, "", "{}"]
+        data["date"] = [7, "", "{}"]
+        data["total_runtime"] = [3, " s", "%.3f"]
+        data["total_wirelength"] = [2, " units", "%.0f"]
+        data["num_clb"] = [4, " blocks", "%.2f"]
+        data["min_chan_width"] = [5, " tracks", "%.3f"]
+        data["crit_path_delay"] = [6, " ns", "%.3f"]
+        table = PrettyTable()
+        table.field_names = list(data.keys())
+        results.readline()
+        for line in results.readlines():
+            info = line.split()
+            row = []
+            for _, values in data.items():
+                if len(info) - 1 < values[0]:
+                    row += [""]
+                else:
+                    if values[2] == "{}" or not info[values[0]].isnumeric():
+                        row += [("{}".format(info[values[0]])) + values[1]]
+                    else:
+                        row += [(values[2] % float(info[values[0]])) + values[1]]
+            table.add_row(row)
+        print(table)
+    return 0
+
+
+def run_odin_test(args, test_name):
+    """ Run ODIN II test with given test name """
+    odin_reg_script = [
+        find_vtr_file("verify_odin.sh"),
+        "--clean",
+        "-C",
+        find_vtr_file("output_on_error.conf"),
+        "--nb_of_process",
+        str(args.j),
+        "--test",
+        "{}/ODIN_II/regression_test/benchmark/".format(find_vtr_root()),
+    ]
+    if test_name == "odin_reg_full":
+        odin_reg_script[-1] += "suite/full_suite"
+    elif test_name == "odin_reg_syntax":
+        odin_reg_script[-1] += "task/syntax"
+    elif test_name == "odin_reg_arch":
+        odin_reg_script[-1] += "task/arch_sweep"
+    elif test_name == "odin_reg_operators":
+        odin_reg_script[-1] += "task/operators"
+    elif test_name == "odin_reg_large":
+        odin_reg_script[-1] += "task/large"
+    elif test_name == "odin_reg":
+        odin_reg_script[-1] += "task/full"
+    elif test_name == "odin_reg_micro":
+        odin_reg_script[-1] += "suite/light_suite"
+
+    odin_root = str(Path(odin_reg_script[0]).resolve().parent)
+
+    result = subprocess.call(odin_reg_script, cwd=odin_root)
+
+    assert result is not None
+    if result != 0:
+        # Error
+        print("FAILED test '{}'".format(test_name))
+        return 1
+
+    # Pass
+    print("PASSED test '{}'".format(test_name))
+    return 0
+
+
+def collect_task_list(reg_test):
+    """ create a list of task files """
+    vtr_task_list_files = []
+    if reg_test.startswith("vtr"):
+        task_list_filepath = str(
+            Path(find_vtr_root())
```

Comment:
Can this path be an argument or environment variable? At the least, please put arbitrary constants (such as a location which could change) at the top of the file with a description.

Reply:
An environment variable is used in the function find_vtr_root() I updated this to use it more efficiently. Let me know if there is anything else I should do with this.